src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -223,44 +227,87 @@ private ConstantValue DefaultSyntaxValue
             }
         }
 
-        // If binder is null, then get it from the compilation. Otherwise use the provided binder.
-        // Don't always get it from the compilation because we might be in a speculative context (local function parameter),
-        // in which case the declaring compilation is the wrong one.
-        protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder binder)
+        private Binder GetBinder(SyntaxNode syntax)
+        {
+            var binder = ParameterBinderOpt;
+
+            // If binder is null, then get it from the compilation. Otherwise use the provided binder.
+            // Don't always get it from the compilation because we might be in a speculative context (local function parameter),
+            // in which case the declaring compilation is the wrong one.
+            if (binder == null)
+            {
+                var compilation = this.DeclaringCompilation;
+                var binderFactory = compilation.GetBinderFactory(syntax.SyntaxTree);
+                binder = binderFactory.GetBinder(syntax);
+            }
+            Debug.Assert(binder.GetBinder(syntax) == null);
+            return binder;
+        }
+
+        private void NullableAnalyzeParameterDefaultValueFromAttributes(DiagnosticBag diagnostics)
         {
+            var defaultValue = DefaultValueFromAttributes;
```

Comment:
Consider skipping all this work if we don't need to do nullable analysis.

Reply:
Happy to do this in the follow-up at the same time as moving this check to a more appropriate spot.